libs/server/Objects/Set/SetObjectImpl.cs

```diff
@@ -256,5 +258,112 @@ private void SetPop(byte* input, int length, ref SpanByteAndMemory output)
                 output.Length = (int)(curr - ptr);
             }
         }
+
+        private void SetRandomMember(byte* input, int length, ref SpanByteAndMemory output)
+        {
+            var _input = (ObjectInputHeader*)input;
+            int count = _input->count;
+
+            byte* input_startptr = input + sizeof(ObjectInputHeader);
+            byte* input_currptr = input_startptr;
+
+            int countDone = 0;
+            bool isMemory = false;
+            MemoryHandle ptrHandle = default;
+            byte* ptr = output.SpanByte.ToPointer();
+
+            var curr = ptr;
+            var end = curr + output.Length;
+
+            ObjectOutputHeader _output = default;
+
+            try
+            {
+                int[] indexes = default;
+
+                if (count > 0)
+                {
+                    // Return an array of distinct elements
+                    var countParameter = count > set.Count ? set.Count : count;
+
+                    // The order of fields in the reply is not truly random
+                    indexes = new List<int>(Enumerable.Range(0, set.Count).OrderBy(x => Guid.NewGuid()).Take(countParameter)).ToArray();
```

Comment:
nit: You don't need a list here either:

Enumerable.Range(0, set.Count).OrderBy(x => Guid.NewGuid()).Take(countParameter).ToArray()

That way you're only enumerating the collection once and only allocating memory once instead of twice